terraform/modules/gcp/secret-manager/main.tf

```diff
@@ -0,0 +1,26 @@
+data "google_project" "project" {}
+
+resource "google_service_account" "service_account" {
+  account_id   = "e2e-test-user"
+  display_name = "KEDA e2e test user"
+  project = data.google_project.project.project_id
+}
+
+resource "google_service_account_key" "credentials" {
+  service_account_id = google_service_account.service_account.name
+}
+
+provider "google" {
+  credentials = google_service_account_key.my_service_account_key.private_key
+  project     = data.google_project.project.project_id
+}
```

Comment:
We don't need top create another service account because KEDA already has its own service account with owner role (to not deal with access problems, totally outside KEDA's scope):
https://github.com/kedacore/testing-infrastructure/blob/main/terraform/modules/gcp/iam/main.tf#L3-L10

Reply:
Got it, so I should remove everything from Line 3 to 16, right?

Reply:
I think that 3 to 11 (13-16 it's the provider), but wait because if we can setup the secrets during the tests, we can remove the whole module and just keep the enabling of the API

Reply:
Removing the entire module as we will be storing the secret as part of the test.

Reply:
yeah, that's the best option. Let's just enable the API and set the secrets during the test